cdci_osa_plugin/osa_dataserve_dispatcher.py

```diff
@@ -657,14 +657,20 @@ class OsaQuery(ProductQuery):
     
     def check_query_roles(self, provided_roles: List[str], par_dic: dict):
         # TODO: there is this get_par_by_name there, to check if it can be made work
-        
+        # the _parameters_list for the query should be set
         max_pointings = int(par_dic.get('max_pointings', 50))
+        # max_pointings = self.get_par_by_name('max_pointings').value
         scw_list = par_dic.get('scw_list', '')
+        # scw_list = self.get_par_by_name('scw_list').value
+        if isinstance(scw_list, str):
+            scw_list = scw_list.split(",")
+
         integral_data_rights = par_dic.get('integral_data_rights', 'public')
+        # integral_data_rights = self.get_par_by_name('integral_data_rights').value
 
         needed_roles = []
 
-        if max_pointings > 50 or len(scw_list.split(",")) > 50:
+        if max_pointings > 50 or len(scw_list) > 50:
```

Comment:
are you sure it is not a string? should be checked by the way

Reply:
Few lines above I check it, and in case, do the split

Reply:
ok, good